### PR TITLE
HTCONDOR-1968: Warn about old job router configuration syntax depreca…

### PR DESCRIPTION
**Pull request by ColeBollig**
…tion

-Added warning message to all provided configuration files
 at the use of one of the deprecated macros
-In verify_ce_config.py if using the old configuration display
 warning message

### CODE REVIEW

File: config/01-ce-router-defaults.conf [hunk 1]
```diff
@@ -106,6 +106,14 @@ SCHEDD_ATTRS = $(SCHEDD_ATTRS) HTCondorCEVersion grid_resource
...
 #
 MERGE_JOB_ROUTER_DEFAULT_ADS=True
+#!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+# Warning: JOB_ROUTER_DEFAULTS, JOB_ROUTER_ENTRIES, JOB_ROUTER_ENTRIES_CMD, and
+#          JOB_ROUTER_ENTRIES_FILE are deprecated and will be removed for V24 of
+#          HTCondor. New configuration syntax for the job router is defined using
+#          JOB_ROUTER_ROUTE_NAMES and JOB_ROUTER_ROUTE_<name>. For new syntax example visit:
+# https://htcondor.readthedocs.io/en/latest/grid-computing/job-router.html#an-example-configuration
+#   Note: The removal will occur during the lifetime of the HTCondor V23 feature series.
+#!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```
Comment (brianhlin) [inline config/01-ce-router-defaults.conf:116]: "will be removed for V24 of HTCondor" should be reworded to one of the following:

- "will be removed for V24 of HTCondor and HTCondor-CE"
- "will be removed for V24 of the HTCondor Software Suite"

File: src/verify_ce_config.py [hunk 2]
```diff
@@ -78,6 +78,16 @@ def main():
...
     else:
+        used_deprecated_knobs = []
+        for knob in ['JOB_ROUTER_DEFAULTS', 'JOB_ROUTER_ENTRIES', 'JOB_ROUTER_ENTRIES_CMD','JOB_ROUTER_ENTRIES_FILE']:
+            if knob in htcondor.param:
+                used_deprecated_knobs.append(knob)
+        if len(used_deprecated_knobs) > 0:
+            warn("%s are deprecated and will be removed for V24 of HTCondor. New configuration"
+                 % ", ".join(used_deprecated_knobs)
+                 + " syntax for the job router is defined using JOB_ROUTER_ROUTE_NAMES and JOB_ROUTER_ROUTE_<name>."
+                 + " Note: The removal will occur during the lifetime of the HTCondor V23 feature series.\n")
```
Comment (brianhlin) [inline src/verify_ce_config.py:89]: A link to the new syntax would here would be great

File: src/verify_ce_config.py [hunk 3]
```diff
@@ -78,6 +78,16 @@ def main():
 
     # If no JOB_ROUTER_ROUTE_<name> rules exist, verify JOB_ROUTER_DEFAULTS and JOB_ROUTER_ENTRIES
     else:
+        used_deprecated_knobs = []
+        for knob in ['JOB_ROUTER_DEFAULTS', 'JOB_ROUTER_ENTRIES', 'JOB_ROUTER_ENTRIES_CMD','JOB_ROUTER_ENTRIES_FILE']:
+            if knob in htcondor.param:
+                used_deprecated_knobs.append(knob)
+        if len(used_deprecated_knobs) > 0:
+            warn("%s are deprecated and will be removed for V24 of HTCondor. New configuration"
+                 % ", ".join(used_deprecated_knobs)
```
Comment (brianhlin) [inline src/verify_ce_config.py:87]:
```suggestion
            warn(f"{', '.join(used_deprecated_knobs) are deprecated and will be removed for V24 of HTCondor. New configuration"
```

I'm partial to f-strings

**Review by brianhlin — APPROVED** [review]
Mostly LGTM. See in-line comments. Pre-approving

**Comment by ColeBollig** [comment]
Here are those minor anticipated changes.

File: src/verify_ce_config.py [hunk 4]
```diff
@@ -78,6 +78,16 @@ def main():
 
     # If no JOB_ROUTER_ROUTE_<name> rules exist, verify JOB_ROUTER_DEFAULTS and JOB_ROUTER_ENTRIES
     else:
+        used_deprecated_knobs = []
+        for knob in ['JOB_ROUTER_DEFAULTS', 'JOB_ROUTER_ENTRIES', 'JOB_ROUTER_ENTRIES_CMD','JOB_ROUTER_ENTRIES_FILE']:
+            if knob in htcondor.param:
+                used_deprecated_knobs.append(knob)
+        if len(used_deprecated_knobs) > 0:
+            warn(f"{', '.join(used_deprecated_knobs)} are deprecated and will be removed for V24 of HTCondor. New configuration"
```
Comment (brianhlin) [inline src/verify_ce_config.py:86]:
```suggestion
            warn(f"{', '.join(used_deprecated_knobs)} are deprecated and will be removed for V24 of the HTCondor Software Suite. New configuration"
```

Oops, missed one

**Review by brianhlin — APPROVED** [review]
LGTM

**Comment by ColeBollig** [comment]
@timtheisen said that the EL8 test failures are not related to the given changes so I will merge this

**Pull request merged**